### ISSUE TITLE
Death test fails on system with specific locales on Windows

### ISSUE THREAD

**Issue by egorpugin** [google/googletest#3196]
Hi,

Death test fails for me because of the line https://github.com/google/googletest/blob/master/googletest/src/gtest-death-test.cc#L765
In my locale (RU) numbers have grouping of 3: `123 456 789`.
On that line process id (win32) is printed like `1 234` or `12 345`.
Child program cannot parse it with space on the line https://github.com/google/googletest/blob/master/googletest/src/gtest-death-test.cc#L1606

Obvious easy fix is to eat spaces in pid string on the line https://github.com/google/googletest/blob/master/googletest/src/gtest-death-test.cc#L1606
```
  fields[3].erase(std::remove_if(fields[3].begin(), fields[3].end(), isspace), fields[3].end());
```
Proper fix is to eat all spaces on the parent side.

Wrong (?) fix is to set C locale always for all tests. This is probably wrong, because we want to test programs in any locale (including current one).
Or we can add ability to set gtest locale using command line option.

**Comment by egorpugin** [comment]
Even more. The space between parts of pid has value `-96` instead of `32`. Looks like something locale specific.
![image](https://user-images.githubusercontent.com/4325333/103322256-812c4a00-4a4e-11eb-96b3-5a55bced7682.png)

**Comment by egorpugin** [comment]
Works for me.
```
  for (auto &c : fields[3]) if (c < 0) c = ' ';
  fields[3].erase(std::remove_if(fields[3].begin(), fields[3].end(), isspace), fields[3].end());
```

**Comment by egorpugin** [comment]
Happened for the `fields[4]` also. Then it's possible for `fields[5]` too.

**Comment by egorpugin** [comment]
Fixed with settings C locale.

**Comment by stweil** [comment]
> Wrong (?) fix is to set C locale always for all tests. This is probably wrong, because we want to test programs in any locale (including current one).

Yes, depending on the C locale is not the right fix for the reasons which you have named here. So if there is a problem in gtest which is caused by locale settings, this issue should be re-opened.
